src/util/submit-addon.js

```diff
@@ -293,6 +294,11 @@ export default class Client {
       [versionObject]: {id: newVersionId},
     } = await this.doNewAddonSubmit(uploadUuid, metaDataJson);
 
+    await saveIdFunc(addonId);
+    log.info(`Generated extension ID: ${addonId}.`);
+    log.info('You must add the following to your manifest:');
+    log.info(`"browser_specific_settings": {"gecko": "${addonId}"}`);
```

Comment:
If we move `saveIdToSourceDir()` somewhere else (could be in this file even), then those logs could also be moved to `saveIdToSourceDir()`. If there is a chance that this message does not apply, we could update the content a bit (e.g. from "You must add [...]" to "Make sure you have [...] in your manifest:").

Reply:
The `id` is never needed for the `signAddon` route, while AMO doesn't allow MV3 submissions. (If we're specifically adding support for MV3 I'd prefer that in a separate PR).

We only call the function for `id`-less submissions for `submit-addon`, so it's always needed.